torchsynth/synth.py

```diff
@@ -214,15 +214,17 @@ def randomize(self, seed: Optional[int] = None):
         Randomize all parameters
         """
         if seed is not None:
-            # Profile to make sure this isn't too slow?
-            mt19937_gen = csprng.create_mt19937_generator(seed)
+            torch.manual_seed(seed)
             for parameter in self.parameters():
                 if not ModuleParameter.is_parameter_frozen(parameter):
-                    parameter.data.uniform_(0, 1, generator=mt19937_gen)
+                    new_params = torch.rand(
+                        (self.batch_size,), device="cpu", pin_memory=True
+                    )
```

Comment:
torch rand was faster by 150ms over 64 iterations at batch size of 64. It's small but adds up to ~10hrs on Synth1B1